app/collections/show.cjsx

```diff
@@ -36,25 +40,38 @@ CollectionPage = React.createClass
 
   render: ->
     <PromiseRenderer promise={@props.collection.get('owner')}>{(owner) =>
-      baseLink = "/collections/#{@props.collection.slug}"
+      baseType = if @props.collection.favorite then "favorites" else "collections"
+      baseLink = ""
+      if @props.project?
+        baseLink = "/projects/#{@props.project.slug}"
+      baseCollectionLink = "#{baseLink}/#{baseType}/#{@props.collection.slug}"
+      baseCollectionsLink = "#{baseLink}/#{baseType}/#{owner.login}"
       isOwner = @props.user?.id is owner.id
+      profileLink = "#{baseLink}/users/#{owner.login}"
+      collectionsLinkMessageKey = "collectionPage.#{baseType}Link"
 
       <div className="collections-page">
         <nav className="collection-nav tabbed-content-tabs">
-          <IndexLink to="#{baseLink}" activeClassName="active" className="tabbed-content-tab" onClick={@logClick?.bind(this, 'view-collection')}>
+          <IndexLink to="#{baseCollectionLink}/#" activeClassName="active" className="tabbed-content-tab" onClick={@logClick?.bind(this, 'view-collection')}>
             <Avatar user={owner} />
             {@props.collection.display_name}
           </IndexLink>
 
           {if isOwner
-            <Link to="#{baseLink}/settings" activeClassName="active" className="tabbed-content-tab" onClick={@logClick?.bind(this, 'settings-collection')}>
+            <Link to="#{baseCollectionLink}/settings" activeClassName="active" className="tabbed-content-tab" onClick={@logClick?.bind(this, 'settings-collection')}>
               <Translate content="collectionPage.settings" />
             </Link>}
 
           {if isOwner
-            <Link to="#{baseLink}/collaborators" activeClassName="active" className="tabbed-content-tab" onClick={@logClick?.bind(this, 'collab-collection')}>
+            <Link to="#{baseCollectionLink}/collaborators" activeClassName="active" className="tabbed-content-tab" onClick={@logClick?.bind(this, 'collab-collection')}>
               <Translate content="collectionPage.collaborators" />
             </Link>}
```

Comment:
👍 

Was getting carried away with the whole "only one surrounding component" thing but I guess that only applies when returning from a `render` function.